src/puptoo/app.py

```diff
@@ -224,8 +224,7 @@ def handle_message(msg, service):
             if facts["system_profile"].get("yum_updates"):
                 if msg.get("custom_metadata") is None:
                     msg["custom_metadata"] = {}
-                msg["custom_metadata"]["yum_updates"] = facts["system_profile"]["yum_updates"]
-                yum_updates = msg["custom_metadata"]["yum_updates"]
+                yum_updates = facts["system_profile"].get("yum_updates")
```

Comment:
```suggestion
                yum_updates = facts["system_profile"].get("yum_updates")
                # delete yum_updates from system_profile as it is already present under custom_metadata 
                del facts["system_profile"]["yum_updates"]
```

Do we need this `if` block that checks for `custom_metadata`? 
I think, when `facts["system_profile"].get("yum_updates")` present, you can directly get `yum_updates` value and remove it from `facts["system_profile"]`. correct?

Reply:
It's not necessary we get `custom_metadata` field.

Reply:
> It's not necessary that we always get `custom_metadata` field.

@r14chandra  - That means no need of `if` block. could you please remove it?

Reply:
If it's not present we need to add that field.

Reply:
> If it's not present we need to add that field.

From this `if` condition, it is difficult to get context. Will it better if you handle this `None` condition above this line - https://github.com/RedHatInsights/insights-puptoo/blob/master/src/puptoo/upload.py#L21?

Reply:
> > If it's not present we need to add that field.
> 
> From this `if` condition, it is difficult to get context. Will it better if you handle this `None` condition above this line - https://github.com/RedHatInsights/insights-puptoo/blob/master/src/puptoo/upload.py#L21?

Apart from `yum_updates_url`, are there any keys that puptoo set under `custom_metadata`.
If yes, then we can keep it here inside app.py file and it would be good to have note why puptoo assigned empty dict to `custom_metadata` if it's `None`.